int8 mo
int8 momsn
int8 mt
int8 mtmsn
int8 waiting
